x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/blob/BlobStoreCacheService.java

```diff
@@ -314,7 +271,7 @@ public ByteRange computeBlobCacheByteRange(String fileName, long fileLength, Byt
             }
         }
 
-        if (METADATA_FILES_EXTENSIONS.contains(fileExtension)) {
+        if (fileExtension.isMetadata()) {
```

Comment:
This would lead to a NPE in case where we have an unknown extension - not good. Let's safeguard against this.

Reply:
:facepalm: thanks for spotting this